File: leetcode/67/addBinary.java
class AddBinary {
    public static void main(String[] args) {
        Solution sl = new Solution();
        System.out.println(sl.addBinary("11", "1"));
    }
}

class Solution {
    public String addBinary(String a, String b) {
        StringBuilder result = new StringBuilder();
        int n1 = a.length() - 1, n2 = b.length() - 1, carry = 0;

        while (n1 >= 0 || n2 >= 0 || carry > 0) {
            int digit1 = (n1 >= 0) ? (a.charAt(n1) - '0') : 0;
            int digit2 = (n2 >= 0) ? (b.charAt(n2) - '0') : 0;

            int sum = digit1 + digit2 + carry;

            result.insert(0, sum % 2);
            carry = sum / 2;

            n1--;
            n2--;
        }

        return result.toString();
    }
}

File: leetcode/67/addBinary.js
var addBinary = function(a, b) {
    let index_a = a.length-1,index_b = b.length-1, carry = 0, result= '', num1 = 0, num2 = 0, sum = 0;
    
    while(index_a>=0|| index_b>=0|| carry ===1)
    {
        num1 = index_a >=0 ? a[index_a--] - '0' : 0;
        num2 = index_b >=0 ? b[index_b--] - '0' : 0;
        
        
        sum = num1 + num2 + carry;
        result = ((sum)%2).toString() + result ;

        console.log(num1,num2,sum,carry,result);

        carry = Math.floor((sum)/2);
    }
    return result;
};

console.log(addBinary('1010','1011'));